content/docs/for-developers/sending-email/automating-subusers.md

```diff
@@ -0,0 +1,513 @@
+"data": "s2.domainkey.u624781.wl.sendgrid.net"
+},
+"spf": {
+"valid": false,
+"type": "txt",
+"host": "staples.com",
+"data": "v=spf1 include:u624781.wl.sendgrid.net -all"
+}
+}
+```
+
+You will want to look for the domain Whitelabel ID here, as this will be used in the next calls to append this Whitelabel to Subusers. Also, the records needed to validate this domain Whitelabel will be passed back in the response.
+
+2. **Create an IP Whitelabel**
+
+```bash
+curl -X POST -H "Authorization: Basic XXXXXXXXXXXXXXX" -H "Content-Type: application/json” -d '{"ip": "167.89.66.50",
+  "subdomain": "email",
+  "domain": "staples.com"}' 'https://api.sendgrid.com/v3/whitelabel/ips'
+ ```
+  
+  Successful Response: 
+ ```bash
+ => 2xx
+  {
+  "id": 11887,
+  "ip": "167.89.66.50",
+  "rdns": "o4.email.staples.com",
+  "users": [],
+  "subdomain": "email",
+  "domain": "staples.com",
+  "a_record": {
+  "valid": false,
+  "type": "a",
+  "host": "o4.email.staples.com",
+  "data": "167.89.66.50"
+  },
+  "valid": false,
+  "legacy": false
+  }
+```
+
+3. **Create link Whitelabel**
+
+```bash
+curl -X POST -H "Authorization: Basic XXXXXXXXXXXXXX" -H "Content-Type: application/json" -d '{
+  "domain": "staples.com",
+  "subdomain": "links",
+  "default": true
+  }' 'https://api.sendgrid.com/v3/whitelabel/links'
+```
+
+Response:
+```bash
+=> 2xx
+{
+"id": 42371,
+"user_id": 624781,
+"domain": "staples.com",
+"subdomain": "links",
+"username": "ryan.burrer",
+"valid": false,
+"default": true,
+"legacy": false,
+"dns": {
+"domain_cname": {
+"valid": false,
+"type": "cname",
+"host": "links.staples.com",
+"data": "sendgrid.net"
+},
+"owner_cname": {
+"valid": false,
+"type": "cname",
+"host": "624781.staples.com",
+"data": "sendgrid.net"
+}
+}
+}
+```
+After creating DNS records then wait for them to propagate and validate records.
+
+4. **GET all domain Whitelabels** - grab the ID to validate and associate to an account.
+
+```bash
+curl -X GET -H "Authorization: Basic XXXXXXXXXXXXXXXX" -H "Content-Type: application/json" 'https://api.sendgrid.com/v3/whitelabel/domains'
+```
+
+Response:
+
+```bash
+= 2xx ok
+{
+"id": 50784,
+"user_id": 624781,
+"subdomain": "subdomain",
+"domain": "staples.com",
+"username": "ryan.burrer",
+"ips": [
+"167.89.38.39",
+"198.37.152.144"
+],
+"custom_spf": false,
+"default": true,
+"legacy": false,
+"automatic_security": true,
+"valid": false,
+"dns": {
+"mail_cname": {
+"valid": false,
+"type": "cname",
+"host": "subdomain.staples.com",
+"data": "u624781.wl.sendgrid.net"
+},
+"dkim1": {
+"valid": false,
+"type": "cname",
+"host": "s1._domainkey.staples.com",
+"data": "s1.domainkey.u624781.wl.sendgrid.net"
+},
+"dkim2": {
+"valid": false,
+"type": "cname",
+"host": "s2._domainkey.staples.com",
+"data": "s2.domainkey.u624781.wl.sendgrid.net"
+},
+"spf": {
+"valid": false,
+"type": "txt",
+"host": "staples.com",
+"data": "v=spf1 include:u624781.wl.sendgrid.net -all"
+}
+}
+}
+```
+
+5. **Validate ID of domain Whitelabel**
```

Comment:
Add another line between this and the one above. Currently running into one another on one line.